src/acl/Ip.cc

```diff
@@ -252,19 +252,117 @@ acl_ip_data::FactoryParse(const char *t)
         return q;
     }
 
-    /* Special ACL RHS "ipv6" matches IPv6-Unicast Internet */
+    /* Special ACL RHS "ipv6" matches IPv6-only Internet addresses */
     if (strcmp(t, "ipv6") == 0) {
         debugs(28, 9, "aclIpParseIpData: magic 'ipv6' found.");
         r = q; // save head of the list for result.
 
-        /* 0000::/4 is a mix of localhost and obsolete IPv4-mapping space. Not valid outside this host. */
+        /* RFC 4291 section 2.5.2. The Unspecified Address: ::/128 */
+        // shared with IPv4 0.0.0.0/32
+
+        /* RFC 4291 section 2.5.3. The Loopback Address : ::1/128 */
+        // shared with IPv4 127.0.0.1/32
+
+        /* RFC 4291 IPv4-compatible IPv6 addres (deprecated): ::/96 */
+        // shared with IPv4 0.0.0.0/0
+
+        /* undefined: default global unicast space with 000* binary prefix */
+        q->next = new acl_ip_data;
+        q = q->next;
```

Comment:
This PR code leaves the initial `q` node (the one returned to the caller as `r`) "empty". This code also mismatches similar leading blobs in "all"- and "ipv4"-handling statements. AFAICT, 62a20b8 went one blob too far:

```suggestion
```